zio-http/shared/src/main/scala/zio/http/codec/SegmentCodec.scala

```diff
@@ -439,9 +439,12 @@ object SegmentCodec          {
     def matches(segments: Chunk[String], index: Int): Int = {
       if (index < 0 || index >= segments.length) -1
       else {
-        val lastIndex = inSegmentUntil(segments(index), 0)
-        if (lastIndex == -1 || lastIndex + 1 != segments(index).length) -1
-        else 1
+        try {
+          java.util.UUID.fromString(segments(index))
```

Comment:
This is extremely inefficient. We should avoid throwing an exception that fills the stack trace.

The format of UUIDs is well known, you should instead manually validate that the string is a UUID or not